### PR TITLE
EDSC-3102: Adds temporal tooltips to the timeline

### PR DESCRIPTION
**Pull request by trevorlang**

### CODE REVIEW

File: src/components/TimelineList/TimelineList.scss [hunk 1]
```diff
@@ -70,7 +74,7 @@
     outline: none;
     border: 0;
     box-shadow: 0 0 3px 3px rgba($color__black--050, 0.1);
-    transition: background-color 0.2s ease-in-out;
+    transition: background-color opacity 0.2s ease-in-out;
```
Comment (trevorlang) [inline src/components/TimelineList/TimelineList.scss]: Fix this

File: src/index.js [hunk 2]
```diff
@@ -93,6 +98,18 @@ export const EDSCTimeline = ({
   // The focused interval
   const [focusedInterval, setFocusedInterval] = useState(propsFocusedInterval)
 
+  // Tracks when the temporal range is hovered
+  const [hoveringTemporalRange, setHoveringTemporalRange] = useState(false)
+
+  // Tracks when a temporal marker is hovered
+  const [hoveringTemporalMarker, setHoveringTemporalMarker] = useState(false)
```
Comment (trevorlang) [inline src/index.js]: Look into whether or not this should be a empty string by default

File: src/index.js [hunk 3]
```diff
@@ -510,36 +529,57 @@ export const EDSCTimeline = ({
...
       setTemporalRangeMouseOverPosition(startPosition)
+      setWillCancelTemporalSelection(true)
     } else {
       setTemporalRangeMouseOverPosition(null)
+      setWillCancelTemporalSelection(false)
+    }
+
+    // If the users cursor is over an interval bottom div, we prevent pointer events on
+    // the temporal selection element via a conditional classname.
+    if (intervalBottomHovered) {
```
Comment (trevorlang) [inline src/index.js]: Clean this syntax up

File: src/utils/determineTemporalLabel.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {
```
Comment (trevorlang) [inline src/utils/determineTemporalLabel.js:1]: Comment

File: src/utils/determineTooltipPosition.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { getPositionByTimestamp } from './getPositionByTimestamp'
+
+export const determineTooltipPosition = ({
```
Comment (trevorlang) [inline src/utils/determineTooltipPosition.js:12]: Comment

**Review by macrouch — APPROVED** [review]

**Review by abbottry — APPROVED** [review]
I _hate_ the waits. But I trust ya'll have investigated.

**Pull request merged**